src/Features/Core/Portable/Completion/CompletionItem.cs

```diff
@@ -20,6 +21,8 @@ public sealed class CompletionItem : IComparable<CompletionItem>
     {
         private readonly string? _filterText;
         private string? _lazyEntireDisplayText;
+        private ImmutableDictionary<string, string>? _propertiesAsImmutableDictionary;
```

Comment:
>private ImmutableDictionary<string, string>? _propertiesAsImmutableDictionary; [](http://example.com/codeflow?start=8&length=78)

This file is the real meat of everything. Essentially, I switch over this class to use an ImmutableArray to back the properties collection and fallback to the exposed ImmutableDictionary when the collection size exceeds 10 elements.

Reply:
Can you add the old member to the list of banned apis within roslyn?

Reply:
Is that typically just done via an ObsoleteAttribute, or does this involve modification of a separate file?

Reply:
it's a separate file.  It's called BannedSymbols.txt.  tehre are a few floating around.